### ISSUE TITLE
[no-unused-vars] false positive about exported type definitions

### ISSUE THREAD

**Issue by iwata** [bradzacher/eslint-plugin-typescript#189]
**Repro**
```JSON
{
  "rules": {
    "no-unused-vars": "error",
    "typescript/no-unused-vars": "error"
  }
}
```

```TS
import firebase, {User} from 'firebase/app'

// initialize firebase project
firebase.initializeApp({
  ...
})
export function authenticated(cb: (user: User | null) => void): void {
  firebase.auth().onAuthStateChanged(user => cb(user))
}
```

**Expected Result**
It should lint without errors.

**Actual Result**

```
utils/authentication.ts:1:19
✖    1:19  User is defined but never used.                                                                 no-unused-vars
```

**Versions**

| package                    | version |
| -------------------------- | ------- |
| `eslint-plugin-typescript` | `0.12.0` |
| `typescript-eslint-parser` | `21.0.1` |
| `typescript`               | `3.1.6` |

**Comment by bradzacher** [comment]
I haven't dug into it, but I believe it's related to #152
If it is, then it'll be fixed by the parser upgrade.

**Comment by iwata** [comment]
> If it is, then it'll be fixed by the parser upgrade.

Okay, I'll wait for new version of parser.
